extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/SecurityEventHelper.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.security.spi.runtime;
+
+import jakarta.enterprise.inject.spi.BeanManager;
+
+public record SecurityEventHelper(boolean fireEventOnSuccess, boolean fireEventOnFailure) {
+
+    public static SecurityEventHelper of(SecurityEvent successEvent, SecurityEvent failureEvent, BeanManager beanManager,
+            boolean enabled) {
+        if (enabled) {
+            boolean fireAllSecurityEvents = !beanManager.resolveObserverMethods(SecurityEvent.NOOP).isEmpty();
```

Comment:
I'd suggest moving SecurityEvent.NOOP here instead of keeping it in the interface itself, though I'm wondering if we can avoid dealing with instances

Reply:
> I'd suggest moving SecurityEvent.NOOP here instead of keeping it in the interface itself

Yes, will move it here, agreed. And will do same for other security events.

>  though I'm wondering if we can avoid dealing with instances

no, see answers above